pageindicatorview/src/main/java/com/rd/PageIndicatorView2.java

```diff
@@ -0,0 +1,882 @@
+            return;
+        }
+
+        indicator.setInteractiveAnimation(false);
+        indicator.setLastSelectedPosition(indicator.getSelectedPosition());
+        indicator.setSelectingPosition(position);
+        indicator.setSelectedPosition(position);
+        manager.animate().basic();
+    }
+
+    /**
+     * Set specific circle indicator position to be selected without any kind of animation. If position < or > total count,
+     * accordingly first or last circle indicator will be selected.
+     *
+     * @param position position of indicator to select.
+     */
+    public void setSelected(int position) {
+        Indicator indicator = manager.indicator();
+        AnimationType animationType = indicator.getAnimationType();
+        indicator.setAnimationType(AnimationType.NONE);
+
+        setSelection(position);
+        indicator.setAnimationType(animationType);
+    }
+
+    /**
+     * Clears selection of all indicators
+     */
+    public void clearSelection() {
+        //TODO check
+        Indicator indicator = manager.indicator();
+        indicator.setInteractiveAnimation(false);
+        indicator.setLastSelectedPosition(Indicator.COUNT_NONE);
+        indicator.setSelectingPosition(Indicator.COUNT_NONE);
+        indicator.setSelectedPosition(Indicator.COUNT_NONE);
+        manager.animate().basic();
+    }
+
+    /**
+     * Set progress value in range [0 - 1] to specify state of animation while selecting new circle indicator.
+     *
+     * @param selectingPosition selecting position with specific progress value.
+     * @param progress          float value of progress.
+     */
+    public void setProgress(int selectingPosition, float progress) {
+        Indicator indicator = manager.indicator();
+        if (!indicator.isInteractiveAnimation()) {
+            return;
+        }
+
+        int count = indicator.getCount();
+        if (count <= 0 || selectingPosition < 0) {
+            selectingPosition = 0;
+
+        } else if (selectingPosition > count - 1) {
+            selectingPosition = count - 1;
+        }
+
+        if (progress < 0) {
+            progress = 0;
+
+        } else if (progress > 1) {
+            progress = 1;
+        }
+
+        if (progress == 1) {
+            indicator.setLastSelectedPosition(indicator.getSelectedPosition());
+            indicator.setSelectedPosition(selectingPosition);
+        }
+
+        indicator.setSelectingPosition(selectingPosition);
+        manager.animate().interactive(progress);
+    }
+
+    public void setClickListener(@Nullable DrawController.ClickListener listener) {
+        manager.drawer().setClickListener(listener);
+    }
+
+    private void init(@Nullable AttributeSet attrs) {
+        setupId();
+        initIndicatorManager(attrs);
+
+        if (manager.indicator().isFadeOnIdle()) {
+            startIdleRunnable();
+        }
+
+        pageChangeCallback = new ViewPager2.OnPageChangeCallback() {
```

Comment:
Missing important other override: `fun onPageSelected(position: Int)`
This override should call `onPageSelect(position: Int)` which is currently just going unused in this file.